### PR TITLE
fix(environment) add Proxy to ReflectiveIntrinsicObjectNames

### PR DESCRIPTION
**Pull request by jdalton**
I believe this is all that's needed to avoid wrapping `Proxy`.

### CODE REVIEW

File: src/intrinsics.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ const ReflectiveIntrinsicObjectNames = [
     'EvalError',
     'Function',
     'Object',
+    'Proxy',
```
Comment (manuel-jasso) [inline src/intrinsics.ts:118]: Is there a way to test this?

**Comment by jdalton** [comment]
Commented out the line change and added failing Karma tests. If I uncomment it the Karma tests still fail and cause Jest tests to fail too.

File: test/membrane/reversed-proxy-constructor.spec.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import createSecureEnvironment from "../../lib/browser-realm.js";
+
+describe("Reversed Proxy constructor", () => {
+    it("can be constructed", () => {
+        const evalScript = createSecureEnvironment({
+            endowments: {
+                test({ Proxy }) {
+                    const p = new Proxy({}, {
+                        get() { return 1 }
+                    });
```
Comment (jdalton) [inline test/membrane/reversed-proxy-constructor.spec.js:10]: ☝️ This produces the not a constructor error.

File: test/membrane/reversed-proxy-constructor.spec.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    });
+    it(".revocable() should be supported", () => {
+        const evalScript = createSecureEnvironment({
+            endowments: {
+                test({ Proxy }) {
+                    const { proxy, revoke } = Proxy.revocable({}, {
+                        get() { return 1 }
+                    });
+                    expect(proxy.a).toBe(1);
+                    revoke();
```
Comment (jdalton) [inline test/membrane/reversed-proxy-constructor.spec.js:25]: ☝️ This is weird. `proxy.a` should be `1` but is `undefined` not sure why.

File: src/intrinsics.ts [hunk 4]
```diff
@@ -115,6 +115,7 @@ const ReflectiveIntrinsicObjectNames = [
     'EvalError',
     'Function',
     'Object',
+    // 'Proxy',
```
Comment (caridy) [inline src/intrinsics.ts:118]: yes... just remove the comment :)

File: test/membrane/reversed-proxy-constructor.spec.js (hunk 3 above)
Comment (caridy) [reply test/membrane/reversed-proxy-constructor.spec.js:25]: we will have to figure this out... weird indeed

File: src/intrinsics.ts (hunk 4 above)
Comment (jdalton) [reply src/intrinsics.ts:118]: Ok. I'll remove the comment to expose the other issues.

**Comment by jdalton** [comment]
Now has an error of: `TypeError: Invalid value used as weak map key`

File: test/membrane/reversed-proxy-constructor.spec.js (hunk 3 above)
Comment (caridy) [reply test/membrane/reversed-proxy-constructor.spec.js:25]: did you figure why? code is looking great and tests are passing.

File: src/intrinsics.ts [hunk 5]
```diff
@@ -150,9 +152,11 @@ export function linkIntrinsics(
         const blue = blueIntrinsics[name];
         const red = redIntrinsics[name];
         // new intrinsics might not be available in some browsers, e.g.: AggregateError
-        if (!isUndefined(blue)) {
+        if (isObjectLike(blue)) {
```
Comment (caridy) [inline src/intrinsics.ts:155]: can you provide more details about why this change? the change makes sense... but I wonder which one was the offender here that makes you change the code from defined to is object like?

File: test/membrane/reversed-proxy-constructor.spec.js (hunk 3 above)
Comment (jdalton) [reply test/membrane/reversed-proxy-constructor.spec.js:25]: I have not yet. The jest tests are passing but the karma ones are not (CI just does jest tests)

File: src/intrinsics.ts (hunk 5 above)
Comment (jdalton) [reply src/intrinsics.ts:155]: In the case we really want to know if the value can be a valid weak map key which means more than just `!isUndefined()`. So it's just a more accurate check... the added check for `isObjectLike(blue.prototype)` is for the case of `Proxy` which has no prototype.

**Review by caridy — APPROVED** [review]
LGTM, lets figure the failing test before we merge this.

**Comment by jdalton** [comment]
Ya. It's consistent in fails in both Chrome and Firefox too. So hopefully not browser bugs.

**Comment by jdalton** [comment]
I've got the karma tests passing locally. Still digging to make sure it's the correct solution.

**Comment by jdalton** [comment]
Weird. Tests passing now. Looks all clear. 👍

**Comment by caridy** [comment]
ok, let's roll!
